src/test/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperationsTest.scala

```diff
@@ -3038,6 +3079,118 @@ class TezosDatabaseOperationsTest
+            asof = new Timestamp(3)
+          )
+        )
+
+        val populateAndTest = for {
+          _ <- Tables.AccountsHistory ++= ahrs
+          found <- sut.selectWithPredicates(
+            table = Tables.AccountsHistory.baseTableRow.tableName,
+            columns = List(SimpleField("account_id"), SimpleField("block_id"), SimpleField("asof")),
+            predicates = List.empty,
+            ordering = List(),
+            aggregation = List.empty,
+            temporalPartition = Some("account_id"),
+            snapshot = Some(Snapshot("asof", new Timestamp(2))),
+            outputType = OutputType.json,
+            limit = 10
+          )
+        } yield found
+
+        val result = dbHandler.run(populateAndTest.transactionally).futureValue
+
+        result shouldBe List(
+          Map(
+            "account_id" -> Some("id1"),
+            "block_id" -> Some("blockid2"),
+            "asof" -> Some(new Timestamp(2)),
+            "r" -> Some(1)
+          )
+        )
+      }
+
+      "should get the balance of an account at a specific timestamp" in {
+
+        val ahrs = List(
+          AccountsHistoryRow(
+            accountId = "id1",
+            blockId = "blockid1",
+            balance = BigDecimal(1.0),
+            blockLevel = BigDecimal(1),
+            asof = new Timestamp(1)
+          ),
+          AccountsHistoryRow(
+            accountId = "id2",
+            blockId = "blockid2",
+            balance = BigDecimal(2.0),
+            blockLevel = BigDecimal(2),
+            asof = new Timestamp(2)
+          ),
+          AccountsHistoryRow(
+            accountId = "id3",
+            blockId = "blockid3",
+            balance = BigDecimal(3.0),
+            blockLevel = BigDecimal(3),
+            asof = new Timestamp(3)
+          )
+        )
+
+        val populateAndTest = for {
+          _ <- Tables.AccountsHistory ++= ahrs
+          found <- sut.selectWithPredicates(
+            table = Tables.AccountsHistory.baseTableRow.tableName,
+            columns = List(SimpleField("account_id"), SimpleField("block_id"), SimpleField("asof")),
+            predicates = List.empty,
+            ordering = List(),
+            aggregation = List.empty,
+            temporalPartition = Some("account_id"),
+            snapshot = Some(Snapshot("asof", new Timestamp(2))),
+            outputType = OutputType.json,
+            limit = 10
+          )
+        } yield found
+
+        val result = dbHandler.run(populateAndTest.transactionally).futureValue
+
+        result shouldBe List(
+          Map(
+            "account_id" -> Some("id1"),
+            "block_id" -> Some("blockid1"),
+            "asof" -> Some(new Timestamp(1)),
+            "r" -> Some(1)
+          ),
+          Map(
+            "account_id" -> Some("id2"),
+            "block_id" -> Some("blockid2"),
+            "asof" -> Some(new Timestamp(2)),
+            "r" -> Some(1)
+          )
+        )
+
+      }
```

Comment:
Maybe you might wanna test filtering failures on invalid field choice for the snapshot query field?

Reply:
added test for that